ElectronClient/app/gui/NoteText.jsx

```diff
@@ -897,6 +884,41 @@ class NoteTextComponent extends React.Component {
 		});
 	}
 
+	commandSavePdf() {
+		const path = bridge().showSaveDialog({
+			filters: [{ name: _('PDF File'), extensions: ['pdf']}],
+			defaultPath: safeFilename(this.state.note.title),
+		});
+
+		if (path) {
+			// temporarily change the html in the webview to show the note title
+			const newHtml = this.insertHtmlHeading_(this.lastSetHtml_, this.state.note.title);
+			this.webview_.send('setHtml', newHtml);
+
+			this.webview_.printToPDF({}, (error, data) => {
+				if (error) {
+					bridge().showErrorMessageBox(error.message);
+				} else {
+					shim.fsDriver().writeFile(path, data, 'buffer');
+				}
+
+				// refresh the webview contents, undoing our temporary change
+				this.lastSetHtml_ = '';
+				this.render()
```

Comment:
A bit hacky but indeed probably the best way to refresh the content. Could we use this.forceUpdate() though instead of calling render directly?

Reply:
Done.